src/buildkite_test_collector/pytest_plugin/buildkite_plugin.py

```diff
@@ -36,48 +40,26 @@ def pytest_runtest_logstart(self, nodeid, location):
 
     def pytest_runtest_logreport(self, report):
         """pytest_runtest_logreport hook callback to get test outcome after test call"""
+        logger.debug('hook=pytest_runtest_logreport nodeid=%s when=%s', report.nodeid, report.when)
 
         # This hook is called three times during the lifecycle of a test:
         # after the setup phase, the call phase, and the teardown phase.
-        # Since we want to capture the outcome from the call phase,
-        # we only proceed when this hook is triggered following the call phase.
+        # We capture outcomes from the call phase, or setup phase if it failed
+        # (since setup failures prevent the call phase from running).
         # See: https://github.com/buildkite/test-collector-python/pull/45
-        if report.when != 'call':
-            return
-
-        nodeid = report.nodeid
-        test_data = self.in_flight.get(nodeid)
-        logger.debug('Enter pytest_runtest_logreport for %s', nodeid)
```

Comment:
Do we want to add a similar debug message to `update_test_result`?

Reply:
I mostly wanted to add it to the PyTest hooks themselves…
the `update_test_result` method does debug-log when it takes action…

```
                logger.debug('-> test passed')
                logger.debug('-> test skipped')
                logger.debug('-> test failed: %s', failure_reason)
```

I think it's okay for it to remain silent when it doesn't do anything.

Also I wasn't sure I should even leave in as much debugging as I did.
I added it for my own purposes in the moment, and tried to make it a bit more consistent than it was, but maybe it should be pruned back down a bit. Not sure!